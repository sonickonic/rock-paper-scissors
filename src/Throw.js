import React, { useState, useEffect } from "react";
import styled from "styled-components";

import Chip from "./Chip";
import shapes from "./data";
import Placeholder from "./Placeholder";

const Text = styled.p`
  text-align: center;
  color: var(--white);
  margin-top: 2.5rem;
`;

const Container = styled.div`
  position: relative;
  width: 31.5rem;
  height: 28.3rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
`;

function Throw({ selected }) {
  const [randomNumber, setRandomNumber] = useState(null);

  useEffect(() => {
    setTimeout(() => {
      setRandomNumber(Math.floor(Math.random() * 3));
    }, 1000);
  }, []);

  return (
    <Container>
      <div>
        <Chip shape={selected} />
        <Text>you picked</Text>
      </div>
      <div>
        {randomNumber || randomNumber === 0 ? (
          <Chip shape={shapes[randomNumber]} />
        ) : (
          <Placeholder />
        )}
        <Text>the house picked</Text>
      </div>
    </Container>
  );
}

export default Throw;
