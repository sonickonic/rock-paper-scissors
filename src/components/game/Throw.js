import React from "react";
import styled from "styled-components";

import Hand from "./Hand";
import Placeholder from "./Placeholder";

const Text = styled.p`
  text-align: center;
  color: var(--white);
  margin-top: 2.5rem;
  letter-spacing: 0.16em;

  @media (min-width: 800px) {
    font-size: 2rem;
    margin-bottom: 6rem;
  }
`;

const Container = styled.div`
  position: relative;
  width: 100%;
  max-width: 47rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;

  @media (min-width: 800px) {
    max-width: 60rem;
  }
`;

const HandContainer = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: 800px) {
    flex-direction: column-reverse;
  }
`;

const Throw = ({ userSelectedHand, botSelectedHand, winner }) => {
  return (
    <Container>
      <HandContainer>
        <Hand winner={winner} shape={userSelectedHand} />
        <Text>you picked</Text>
      </HandContainer>
      <HandContainer>
        {botSelectedHand ? (
          <Hand winner={winner} shape={botSelectedHand} />
        ) : (
          <Placeholder />
        )}
        <Text>the house picked</Text>
      </HandContainer>
    </Container>
  );
};

export default Throw;
