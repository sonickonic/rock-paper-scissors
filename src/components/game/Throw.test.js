import React from "react";
import { render } from "@testing-library/react";
import Throw from "./Throw";
import shapes from "../../data";

describe("Throw", () => {
  describe("before bot selected hand", () => {
    const userSelectedHand = shapes[0];

    it("renders correctly", () => {
      const { asFragment } = render(
        <Throw userSelectedHand={userSelectedHand} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it("shows one hand", () => {
      const { container } = render(
        <Throw userSelectedHand={userSelectedHand} />
      );

      expect(container.querySelectorAll("img")).toHaveLength(1);
    });
  });

  describe("after bot selected hand", () => {
    const userSelectedHand = shapes[0];
    const botSelectedHand = shapes[1];

    it("renders correctly", () => {
      const { asFragment } = render(
        <Throw
          userSelectedHand={userSelectedHand}
          botSelectedHand={botSelectedHand}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it("shows two hands", () => {
      const { container } = render(
        <Throw
          userSelectedHand={userSelectedHand}
          botSelectedHand={botSelectedHand}
        />
      );

      expect(container.querySelectorAll("img")).toHaveLength(2);
    });
  });
});
