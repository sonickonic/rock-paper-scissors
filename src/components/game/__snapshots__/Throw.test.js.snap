// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Throw after bot selected hand renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhUy lfNJJV"
  >
    <div>
      <div
        class="sc-AxjAm hUUgft"
      >
        <img
          alt="paper"
          class="sc-AxirZ hctvIa"
          src="icon-paper.svg"
        />
      </div>
      <p
        class="sc-AxhCb fpVQKk"
      >
        you picked
      </p>
    </div>
    <div>
      <div
        class="sc-AxjAm gnslzt"
      >
        <img
          alt="scissors"
          class="sc-AxirZ hctvIa"
          src="icon-scissors.svg"
        />
      </div>
      <p
        class="sc-AxhCb fpVQKk"
      >
        the house picked
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Throw before bot selected hand renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhUy lfNJJV"
  >
    <div>
      <div
        class="sc-AxjAm hUUgft"
      >
        <img
          alt="paper"
          class="sc-AxirZ hctvIa"
          src="icon-paper.svg"
        />
      </div>
      <p
        class="sc-AxhCb fpVQKk"
      >
        you picked
      </p>
    </div>
    <div>
      <div
        class="sc-AxiKw iMJUuS"
      />
      <p
        class="sc-AxhCb fpVQKk"
      >
        the house picked
      </p>
    </div>
  </div>
</DocumentFragment>
`;
